### ISSUE TITLE
@instana/collector at ^1.106.2 depends on optional dependencies unsafely

### ISSUE THREAD

**Issue by ThisIsMissEm** [instana/nodejs-sensor#257]
If you install your node modules with:

```
yarn install --production --ignore-optional --frozen-lockfile
```

Then the resulting installation will not run, as `@instana/collector fails to require the `event-loop-lag` module, which is marked as optional, but in the code, it is required just as `const lag = require('event-loop-lag')(1000);`

This means our production build cannot start if we use the `@instana/collector` package.

**Comment by ThisIsMissEm** [comment]
Furthermore, you currently depend on `node-gyp` for installs, which results in errors if python is not available on the host system, as is the case in both the `node:lts-slim` and `node:lts-alpine` official docker images.

**Comment by basti1302** [comment]
Thanks again for the fix around the faulty `event-loop-lag` module dependency declaration. 

Regarding the other aspect you mention: Where exactly did you encounter a hard dependency on `node-gyp`? All dependencies that are native add-ons and might use node-gyp (`@instana/autoprofile`, `event-loop-stats`, `gcstats.js`,  and `netlinkwrapper`) are declared as optional dependencies precisely for the reason that Python or other build tools are not available on the target system. I think the code that loads those dependencies should always handles the case that they are not installed correctly (in contrast to the code loading event-loop-lag, which was wrongly classified as optional).

So an `npm install` or `yarn install` on a `slim` or `alpine` image will indeed log errors when trying to install those native add-ons, but it will ultimately succeed. What am I missing?

**Comment by basti1302** [comment]
The event-loop-lag issue has been fixed in version 1.106.3.

**Comment by basti1302** [comment]
Closing this as the reported error around `--ignore-optional ` has been fixed and I didn't hear back from you about the hard dependency on `node-gyp` (I still think there is none).
